airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -298,22 +303,27 @@ def get_jobs(self) -> List[Dict]:
 
     def cancel(self) -> None:
         """
-        Cancels current job
+        Cancels or drains current job
         """
         jobs = self._get_current_jobs()
         batch = self._dataflow.new_batch_http_request()
         job_ids = [job['id'] for job in jobs]
         self.log.info("Canceling jobs: %s", ", ".join(job_ids))
-        for job_id in job_ids:
+        for job in jobs:
+            requested_state = (
+                DataflowJobType.JOB_TYPE_STREAMING
```

Comment:
`DataflowJobStatus.JOB_STATE_DRAINED` instead of `DataflowJobType.JOB_TYPE_STREAMING` ?

Reply:
Thanks! Fixed.